website/docs/record_payload.md

```diff
@@ -0,0 +1,97 @@
+---
+title: Record Payload 
+keywords: [hudi, merge, upsert, precombine]
+---
+
+## Record Payload
+
+One of the core features of Hudi is the ability to incrementally upsert data, deduplicate and merge records on the fly.
+Additionally, users can implement their custom logic to merge the input records with the record on storage. Record
+payload is an abstract representation of a Hudi record that allows the aforementioned capability. As we shall see below,
+Hudi provides out-of-box support for different payloads for different use cases, and a new record merger API for
+optimized payload handling. But, first let us understand how record payload is used in the Hudi upsert path.
+
+<figure>
+    <img className="docimage" src={require("/assets/images/upsert_path.png").default} alt="upsert_path.png" />
+</figure>
+
+Figure above shows the main stages that records go through while being written to the Hudi table. In the precombining
+stage, Hudi performs any deduplication based on the payload implementation and precombine key configured by the user.
+Further, on index lookup, Hudi identifies which records are being updated and the record payload implementation tells
+Hudi how to merge the incoming record with the existing record on storage.
+
+### Existing Payloads
+
+#### OverwriteWithLatestAvroPayload
+
+This is the default record payload implementation. It picks the record with the greatest value (determined by calling
+.compareTo() on the value of precombine key) to break ties and simply picks the latest record while merging. This gives
+latest-write-wins style semantics.
+
+#### EventTimeAvroPayload
+
+Some use cases require merging records by event time and thus event time plays the role of an ordering field. This
+payload is particularly useful in the case of late-arriving data. For such use cases, users need to set
+the [payload event time field](/docs/configurations#RECORD_PAYLOAD) configuration.
+
+#### ExpressionPayload
+
+This payload is very useful when you want to merge or delete records based on some conditional expression, especially
```

Comment:
should we remove this from this list. I thought its meant to be used only internally. can anyone directly set expression payload for their table?

Reply:
Didn't know that this is meant to be used internally. Is there a guard like that on payload class config? cc @alexeykudinkin

Reply:
Correct this is only supposed to be used internally (by MERGE INTO)